rclcpp/test/test_executor.cpp

```diff
@@ -67,3 +67,47 @@ TEST_F(TestExecutors, addTemporaryNode) {
   executor.add_node(std::make_shared<rclcpp::Node>("temporary_node"));
   EXPECT_NO_THROW(executor.spin_some());
 }
+
+// Make sure that the spin_until_future_complete works correctly with std::future
+TEST_F(TestExecutors, testSpinUntilFutureComplete) {
+  rclcpp::executors::SingleThreadedExecutor executor;
+  std::future<void> future;
+  rclcpp::FutureReturnCode ret;
+
+  // test success
+  future = std::async([]() {
+    return;
+  });
+  ret = executor.spin_until_future_complete(future, 1s);
+  EXPECT_EQ(rclcpp::FutureReturnCode::SUCCESS, ret);
+
+  // test timeout
+  future = std::async([]() {
+    std::this_thread::sleep_for(20ms);
```

Comment:
Use a bigger number here e.g.: `1s`. That will avoid test flakiness (Windows time slice is ~100ms)